releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+
+#### 4.1.2. Deploying a cluster
+
+Today workload clusters in our Cluster API-based KaaS product are deployed with cluster-$provider apps. That remains the same, with only one minor difference, which is how we specify the version, where instead of specifying App’s `.spec.version` property we MUST specify `release.giantswarm.io/version` label and leave `.spec.version` empty.
+
+So the whole manifest for the cluster-$provider app and its config would look like this:
+
+```yaml
+---
+apiVersion: v1
+data:
+  values: |
+    global:
+      metadata:
+        name: mycluster
+        organization: mycompany
+        description: Production cluster
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
+  labels:
+    giantswarm.io/cluster: mycluster
+  name: mycluster-userconfig
+  namespace: org-mycompany
+---
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+metadata:
+  labels:
+    app-operator.giantswarm.io/version: 0.0.0
+    release.giantswarm.io/version: 25.0.0
+  name: mycluster
+  namespace: org-mycompany
+spec:
+  catalog: cluster
+  kubeConfig:
+    inCluster: true
+  name: cluster-aws
+  namespace: org-mycompany
+  userConfig:
+    configMap:
+      name: mycluster-userconfig
+      namespace: org-mycompany
+  version: ""
+```
+
+After cluster-$provider app is deployed, few things will happen:
+- an App mutating webhook reads the release version, looks up the corresponding release, reads cluster-$provider app version from it and finally it sets App’s `.spec.version` property,
+- after cluster-$provider app is applied successfully, in order to render all templates and set the app version, catalog and depends on properties, Helm will do the following:
+	- it will lookup the App resource (via Helm `lookup` function) and read release version,
+	- then it will lookup the Release resource and read app version, catalog and depends on properties from there, and
+	- finally it will render all templates and apply them.
+
+From this point forward, everything is the same like when all app details were set directly in the Helm chart.
+
+### 4.2. Comparison with the current releases
+
+This section compares the proposed release process with the current one, where we deploy clusters with cluster-$providers apps which have information about app versions.
+
+We cover some of the topics mentioned in section 3. Motivation.
+
+#### 4.2.1. Release identifier and versioning
+
+With the current cluster-$provider apps releases process, the cluster-$providers app version is the release identifier.
+
+With the releases repository, release identifiers would be defined in the repository itself, in the same way like for vintage releases. In this process the cluster-$provider app is one of the release components.
+
+In both cases there is a single release identifier, just defined in different places.
```

Comment:
What do you mean by different places? 🤔

Reply:
Currently the release identifier (release version) is cluster-$provider app release, so you can look up some release e.g. in cluster-aws repo releases.

With the releases repo you would look up some release in the release repo (as cluster-$provider release becomes just one part of the release).